main.py

```diff
@@ -404,48 +405,49 @@ def main():
     if args.infer or args.eval:
         files = args.infer_files if args.infer else args.val_files
         files = gather_files_from_cmdline(files)
-
-        # Get model parameters
-        with h5py.File(files[0], 'r') as f:
-            args.interval_size = f['data'].shape[1]
-            args.batch_size = 1
-
-        if args.infer:
-            args.infer_files = files
-            _logger.debug("Inference data: ", args.infer_files)
-        else:
-            args.val_files = files
-            _logger.debug("Evaluation data: ", args.val_files)
-
-        # setup queue and kick off writer process
-        #############################################################
-        manager = mp.Manager()
-        res_queue = manager.Queue()
-        write_proc = mp.Process(target=writer, args=(args, res_queue))
-        write_proc.start()
-        #############################################################
-
-        ngpus_per_node = torch.cuda.device_count()
-        # WAR: gloo distributed doesn't work if world size is 1.
-        # This is fixed in newer torch version - https://github.com/facebookincubator/gloo/issues/209
-        args.distributed = False if ngpus_per_node == 1 else args.distributed
-
-        worker = infer_worker if args.infer else eval_worker
-        if args.distributed:
-            args.world_size = ngpus_per_node
-            mp.spawn(worker, nprocs=ngpus_per_node, args=(
-                ngpus_per_node, args, res_queue), join=True)
-        else:
-            assert_device_available(args.gpu)
-            args.world_size = 1
-            worker(args.gpu, ngpus_per_node, args, res_queue)
-
-        # finish off writing
-        #############################################################
-        res_queue.put("done")
-        _logger.info("Waiting for writer to finish...")
-        write_proc.join()
-        #############################################################
+        for x in range(0,len(files)):
+            infile = [files[x]]
+            if args.infer:
```

Comment:
Since you brought up the `args` issue, I'm wondering if we can do a little bit of cleanup for it right now. Instead of updating the `infer_files` or `eval_files` argument, how about updating the signature of the `writer` function to take in the corresponding input file path directly in line 426? Then the writer function won't have to do a list index [0] query